### ISSUE TITLE
How to work with BackoffSupervisor

### ISSUE THREAD

**Issue by VladimirLuzhin** [akkadotnet/akka.net#3562]
- Version Akka.NET v1.3.8
- Platform Windows 10

I'm trying to understand out how the **BackoffSupervisor** should work. This is my test application.
```
class Program
    {
        static void Main(string[] args)
        {
            var config = ConfigurationFactory.ParseString(@"
                akka {  
                    stdout-loglevel = DEBUG
                    loglevel = DEBUG
                    log-config-on-start = off  
                    log-dead-letters = 10
                    actor {
                        debug {  
                              receive = on 
                              autoreceive = on
                              lifecycle = on
                        }
                    }
                }");

            using (var system = ActorSystem.Create("TestSystem", config))
            {
                var childs = Child.Props();
                var parent = BackoffSupervisor.Props(
                    Backoff.OnFailure(
                        childs,
                        "myChilds",
                        TimeSpan.FromSeconds(3),
                        TimeSpan.FromSeconds(30),
                        0.5));

                var supervisor = system.ActorOf(parent, "supervisor");                

                while (true)
                {
                    var input = Console.ReadLine();

                    if (input.ToLower() == "exit")
                        break;
                    else
                        supervisor.Tell(input);
                }

                system.Terminate().Wait();
            }
        }
    }

    public class ParentSuperVisor : ReceiveActor
    {
        public static Props Props()
        {
            return Akka.Actor.Props.Create(() => new ParentSuperVisor());
        }
    }

    public class Child : ReceiveActor
    {
        public Child()
        {
            Receive<string>(msg => HandleMessage(msg));
        }

        private void HandleMessage(string msg)
        {
            throw new Exception("Error");
        }

        public static Props Props()
        {
            return Akka.Actor.Props.Create(() => new Child()).WithRouter(new RoundRobinPool(3));
        }
    }
```
In this implementation, if one **Child** actor falls, then all three **childs** actors fall. I need to restart only the actor who fell. Somebody can tell?

Thank you in advance for your answers.

**Comment by Danthar** [comment]
In this case, its because your child is actually a router. The backoffsupervisor has no knowledge about this. It only works with the child props you feed it. So when it detects that the message it tried to send to its child failed. it restarts the child and tries again.
Because your child is actually a Router, the default behavior of a router when it fails, is to restart all its children.

If you want to use a roundrobinpool of actors which are each individually covered by an BackoffSupervisor then that is what you should do. 

So in short, start a roundrobinpool of child actors which are individually wrapped by an BackoffSupervisor.

**Comment by VladimirLuzhin** [comment]
@Danthar thank you for answer.
